### PR TITLE
Update exposed

### PR DESCRIPTION
**Pull request by imykolenko**

### CODE REVIEW

File: integrationtests/build.gradle.kts [hunk 1]
```diff
@@ -61,12 +61,12 @@ dependencies {
     implementation("io.grpc:grpc-netty:$grpcProtobufVersion")
     implementation("net.openhft:chronicle-queue:5.22.28")
```
Comment (imykolenko) [inline integrationtests/build.gradle.kts:62]: `5.22.28` version is from 2022. Newer versions have `ea` (net.openhft:chronicle-queue:5.26ea9) which suggests `early access` probably for evaluation reason. Where can we obtain newer version?

https://central.sonatype.com/artifact/net.openhft/chronicle-queue/versions

File: backend/build.gradle.kts [hunk 2]
```diff
@@ -1,10 +1,9 @@
-import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
-    kotlin("jvm") version "1.9.23"
-    kotlin("plugin.serialization") version "1.9.23"
+    kotlin("jvm") version "2.0.21"
```
Comment (antonyshchenko) [inline backend/build.gradle.kts:3]: 🔥

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/ArchAccount.kt [hunk 3]
```diff
@@ -82,9 +83,18 @@ sealed class AccountSetupState {
 object ArchAccountTable : GUIDTable<ArchAccountId>("arch_account", ::ArchAccountId) {
     val createdAt = timestamp("created_at")
     val updatedAt = timestamp("updated_at").nullable()
-    val utxoId = varchar("utxo_id", 10485760).uniqueIndex()
-    val creationTxId = varchar("creation_tx_id", 10485760).nullable()
-    val encryptedPrivateKey = varchar("encrypted_private_key", 10485760)
+    val utxoId = varchar("utxo_id", 10485760).uniqueIndex().transform(
+        wrap = { BitcoinUtxoId(it) },
```
Comment (antonyshchenko) [inline backend/src/main/kotlin/xyz/funkybit/core/model/db/ArchAccount.kt:87]: Nice! Is this a new feature of Exposed?

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/EntityId.kt [hunk 4]
```diff
@@ -1,20 +1,100 @@
...
+// class EntityIdColumnType<T : EntityId>(val deserialize: (String) -> T) : VarCharColumnType(colLength = 10485760) {
+//    override fun valueFromDB(value: Any): Any {
+//        return when (val v = super.valueFromDB(value)) {
+//            is String -> deserialize(v)
+//            else -> v
+//        }
+//    }
+//
+//    override fun notNullValueToDB(value: Any): Any = value.toString()
+// }
```
Comment (antonyshchenko) [inline backend/src/main/kotlin/xyz/funkybit/core/model/db/EntityId.kt:100]: Let's remove this

**Review by antonyshchenko — APPROVED** [review]
Great job! 👍

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/ArchAccount.kt (hunk 3 above)
Comment (imykolenko) [reply backend/src/main/kotlin/xyz/funkybit/core/model/db/ArchAccount.kt:87]: yes, one of main changes, transformation is moved from entity to DSL. This allows to use value classes directly in queries. 

Types got stricter, but IDs are no longer comparable, needs to be unwrapped.

Upsert API got better, even promises to read value returned from the DB 😜

**Pull request merged**